### ISSUE TITLE
datepattern not working

### ISSUE THREAD

**Issue by arrayown** [fail2ban/fail2ban#1558]
### Environment:
- Fail2Ban version (including any possible distribution suffixes): 0.9.5
- OS, including release name/version: Ubuntu Server 12.04 64-bit
- [x] Some customizations were done to the configuration (provide details below is so): I installed Fail2Ban with Python 3.3

### The issue:
Customized date pattern does not work. I setup the jail to read a log file outside of /var/log (located at: /usr/local/WowzaStreamingEngine/logs/wowzastreamingengine_access.log). The log prints lines that I want to use to block users.

Log excerpt:
```
2016-09-22	22:00:55	UTC	comment	server	INFO	200	-	doing POST request to [http://xxx/api/auth] with contenttype [application/x-www-form-urlencoded] data: action=start&widget=some.widget&username=example&token=alphanumeric&wowza_session_id=585351812&user_ip=xxx.xxx.xxx.xxx	-	-	-	29218.809	-	-	-	-	-	-	-	-	-	-	-	-	-	-	-	-	-	-	-	-	-	-	-	-	-
```

#### Steps to reproduce
1. Tail fail2ban.log file
2. Restart Fail2Ban
3. Wait for Fail2Ban to match entries from log

#### Expected behavior
Fail2Ban should match strings that it finds in the live log.

#### Observed behavior
Fail2Ban acts like nothing is happening.

#### Any additional information
I tried the simplest configs that should basically ALWAYS return true but it won't work. I tried removing the datepattern altogether hoping that Fail2Ban will figure out the timedate format. No luck.

### Configuration, dump and another helpful excerpts
jail.local:
```
[wowza-softddos]   
enabled  = true 
filter   = wowza-softddos
logpath  = /usr/local/WowzaStreamingEngine/logs/wowzastreamingengine_access.log
maxretry = 10
findtime = 10
bantime  = 600
banaction = iptables-allports
```

filter.d/wowza-softddos.conf:
```
[Definition]
failregex = .*user_ip=<HOST>.*$
#failregex = .*doing POST request to.*action=start.*user_ip=<HOST>.*$
# First regex is just to see what's happening, I need second regex to work

[Init]
datepattern = ^%%Y-%%m-%%d     %%H:%%M:%%S     %%Z
# Tried \t instead of actual tabs, tried 2 and 4 spaces instead of tabs -- nothing worked.
```

#### Any customizations done to /etc/fail2ban/ configuration
```
Installed Fail2Ban 0.9.5 with Python 3.3 on Ubuntu 12.04 Precise.
```

#### Relevant parts of /var/log/fail2ban.log file:
_preferably obtained while running fail2ban with `loglevel = 4`_

```
2016-09-23 02:55:28,721 fail2ban.filter         [10838]: DEBUG   Processing line with time:1474588825.0 and ip:111.222.333.444
2016-09-23 02:55:28,721 fail2ban.filter         [10838]: DEBUG   Ignore line since time 1474588825.0 < 1474599328.7215772 - 10
2016-09-23 02:55:29,184 fail2ban.filter         [10838]: DEBUG   Processing line with time:1474588825.0 and ip:111.222.333.444
2016-09-23 02:55:29,185 fail2ban.filter         [10838]: DEBUG   Ignore line since time 1474588825.0 < 1474599329.1852555 - 10
```

#### Relevant lines from monitored log files in question:
I'm out of ideas what else to do. Please help :(

**Comment by sebres** [comment]
I think, datepattern is part of jail, not filter.
BTW. Filter does not process date time at all, it will be e

**Comment by arrayown** [comment]
When I add datepattern in jail, it doesn't seem to be added to the jail's settings:
```
$ fail2ban-client -d
['add', 'wowza-softddos', 'auto']
['set', 'wowza-softddos', 'ignorecommand', '']
['set', 'wowza-softddos', 'addignoreip', '127.0.0.1/8']
['set', 'wowza-softddos', 'usedns', 'warn']
['set', 'wowza-softddos', 'findtime', 10]
['set', 'wowza-softddos', 'bantime', 600]
['set', 'wowza-softddos', 'logencoding', 'auto']
['set', 'wowza-softddos', 'addlogpath', '/usr/local/WowzaStreamingEngine/logs/wowzastreamingengine_access.log', 'head']
['set', 'wowza-softddos', 'maxretry', 10]
['set', 'wowza-softddos', 'addfailregex', '.*doing POST request to.*action=start.*user_ip=(<HOST>).*$']
['set', 'wowza-softddos', 'addaction', 'iptables-allports']
['set', 'wowza-softddos', 'action', 'iptables-allports', 'actioncheck', "<iptables> -n -L <chain> | grep -q 'f2b-<name>[ \\t]'"]
... (other settings for this action)
```

However, when I leave it in filter, this is what I get
```
$ fail2ban-client -d
['add', 'wowza-softddos', 'auto']
['set', 'wowza-softddos', 'bantime', 600]
['set', 'wowza-softddos', 'addlogpath', '/usr/local/WowzaStreamingEngine/logs/wowzastreamingengine_access.log', 'head']
['set', 'wowza-softddos', 'addignoreip', '127.0.0.1/8']
['set', 'wowza-softddos', 'usedns', 'warn']
['set', 'wowza-softddos', 'findtime', 10]
['set', 'wowza-softddos', 'maxretry', 10]
['set', 'wowza-softddos', 'ignorecommand', '']
['set', 'wowza-softddos', 'logencoding', 'auto']
['set', 'wowza-softddos', 'addfailregex', '.*doing POST request to.*action=start.*user_ip=(<HOST>).*$']
['set', 'wowza-softddos', 'datepattern', '^%Y-%m-%d    %H:%M:%S']
['set', 'wowza-softddos', 'addaction', 'iptables-allports']
```

Any other ideas?

Thank you.

**Comment by sebres** [comment]
Sorry for overlook...

If still matter, I've made a little bit better (more stricter) filter for that, which also works:
```
[Definition]
failregex = ^[^\-]+\d+\s+-\s+doing\s+POST\s+request\s+to\s+\[[^]]*\]\s+with\s+contenttype\s+\[[^]]*\] data: action=\S+&user_ip=<HOST>\s+
[Init]
datepattern = ^%%Y-%%m-%%d\s+%%H:%%M:%%S\s*%%Z
```

But I believe that this issue is additionally concerned from same time zone problem as  #1575 (fixed in 0.10th branch within #1583)

So the testing in CET-Timezone on following file (contains your test excerpt with log-line but IP 1.2.3.4):
```
2016-09-22  22:00:55    UTC comment server  INFO    200 -   doing POST request to [http://xxx/api/auth] with contenttype [application/x-www-form-urlencoded] data: action=start&widget=some.widget&username=example&token=alphanumeric&wowza_session_id=585351812&user_ip=1.2.3.4   -   -   -   29218.809   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -   -
```
using current master:
```
$ git checkout master
Switched to branch 'master'

$ PYTHONPATH=. bin/fail2ban-regex -v test-filters/wowza-softddos.txt test-filters/wowza-softddos.conf
...
Failregex: 1 total
|-  #) [# of hits] regular expression
|   1) [1] ^[^\-]+\d+\s+-\s+doing\s+POST\s+request\s+to\s+\[[^]]*\]\s+with\s+contenttype\s+\[[^]]*\] data: action=\S+&user_ip=<HOST>\s+
|      1.2.3.4  Thu Sep 22 22:00:55 2016
...
Lines: 1 lines, 0 ignored, 1 matched, 0 missed
```
using fixed branch:
```
$ git checkout _0.10/fix-datedetector-grave-fix-v2
Switched to branch '_0.10/fix-datedetector-grave-fix-v2'

$ PYTHONPATH=. bin/fail2ban-regex -v test-filters/wowza-softddos.txt test-filters/wowza-softddos.conf
...
Failregex: 1 total
|-  #) [# of hits] regular expression
|   1) [1] ^[^\-]+\d+\s+-\s+doing\s+POST\s+request\s+to\s+\[[^]]*\]\s+with\s+contenttype\s+\[[^]]*\] data: action=\S+&user_ip=<HOST>\s+
|      1.2.3.4  Fri Sep 23 00:00:55 2016
...
Lines: 1 lines, 0 ignored, 1 matched, 0 missed
```

You see `Fri Sep 23 00:00:55 2016` is not equal `Thu Sep 22 22:00:55 2016` (what IMHO is wrong for time zone CET)
